decidim-blogs/app/permissions/decidim/blog/permissions.rb

```diff
@@ -7,15 +7,91 @@ def permissions
         return permission_action unless permission_action.subject == :blogpost || permission_action.subject == :post
 
         if permission_action.scope == :public
+          if permission_action.action.in?([:update, :destroy])
+            toggle_allow(can_manage_post)
+            return permission_action
+          end
+
+          if permission_action.action == :create
+            toggle_allow(can_create_post)
+            return permission_action
+          end
+
           allow!
           return permission_action
         end
 
-        return permission_action if permission_action.scope != :admin
+        return permission_action unless permission_action.scope == :admin
+
+        if permission_action.action.in?([:update, :destroy])
+          toggle_allow(admin_can_manage_post)
+          return permission_action
+        end
 
         allow!
         permission_action
       end
+
+      def post
+        @post ||= context.fetch(:blogpost, nil)
+      end
+
+      def current_component
+        @current_component ||= context.fetch(:current_component, nil)
+      end
+
+      def can_create_post
+        Decidim.module_installed?("initiatives") &&
+          current_component&.participatory_space.is_a?(Decidim::Initiative) &&
+          initiative_authorship? &&
+          current_component&.participatory_space&.published? &&
+          current_component&.published?
+      end
+
+      def can_manage_post
+        return false unless post&.author
+
+        can_create_post && post&.author == user
+      end
+
+      def admin_can_manage_post
+        return false unless post&.author
+
+        case post.author
+        when Decidim::User
+          post.author == user
+        when Decidim::UserGroup
+          post.author.users.include?(user)
+        when Decidim::Organization
+          space_admin?
+        else
+          false
+        end
+      end
+
+      def space_admin?
+        space_admins.include?(user)
+      end
+
+      def space_admins
+        participatory_space = current_component&.participatory_space
+
+        return [] unless participatory_space
+
+        @space_admins ||= begin
+          space_admins = if participatory_space.respond_to?(:user_roles)
+                           participatory_space.user_roles(:admin)&.collect(&:user)
+                         else
+                           []
+                         end
+          global_admins = current_component.organization.admins
+          (global_admins + space_admins).uniq
+        end
+      end
+
+      def initiative_authorship?
+        current_component&.participatory_space&.has_authorship?(user)
+      end
```

Comment:
This method actually raises an error when you try to view a blogs component with an user that is not logged in: 
![image](https://github.com/user-attachments/assets/d70486b7-a03d-4e33-82ff-979728b29ff1)

```suggestion
      def initiative_authorship?
        return false unless user
        current_component&.participatory_space&.has_authorship?(user)
      end
```